src/openpgp.js

```diff
@@ -78,27 +79,26 @@ if (globalThis.ReadableStream) {
  * @async
  * @static
  */
-export function generateKey({ userIds = [], passphrase = "", rsaBits = null, keyExpirationTime = 0, curve = "curve25519", date = new Date(), subkeys = [{}] }) {
+export async function generateKey({ userIds = [], passphrase = "", rsaBits = null, keyExpirationTime = 0, curve = "curve25519", symmetric = null, date = new Date(), subkeys = [{}] }) {
   userIds = toArray(userIds);
-  curve = rsaBits ? "" : curve;
-  const options = { userIds, passphrase, rsaBits, keyExpirationTime, curve, date, subkeys };
+  symmetric = rsaBits ? "" : symmetric;
```

Comment:
The order these parameters are checked here is meant to preserve the existing defaults. I'm open to shuffling it around if it makes more sense to do it another way.